### PR TITLE
hidpp20: first parsing of feature 0x8070 - onboard profiles

### PR DESCRIPTION
**Pull request by amfern**

### CODE REVIEW

**Comment by amfern** [comment]
following the comments on #113 splitting changes to hidpp20 away from public

File: src/hidpp20.h [hunk 1]
```diff
@@ -117,6 +117,13 @@ enum hidpp20_battery_status {
 	BATTERY_STATUS_INVALID,
 };
 
+enum hidpp20_led_mode {
+	HIDPP20_LED_OFF = 0,
+	HIDPP20_LED_ON = 1,
+	HIDPP20_LED_CYCLE = 03,
```
Comment (whot) [inline src/hidpp20.h]: 03 in octal notation? That seems odd. Either decimal or switch all to hex please

File: src/hidpp20.c [hunk 2]
```diff
@@ -1776,6 +1777,29 @@ hidpp20_buttons_from_cpu(struct hidpp20_profile *profile,
 	}
 }
 
+static void
+hidpp20_onboard_profile_read_led(struct hidpp20_led *led,
+			struct hidpp20_internal_led internal_led)
+{
+	uint16_t rate = 0;
+	uint8_t brightness = 0;
+	switch (internal_led.mode) {
```
Comment (whot) [inline src/hidpp20.c]: empty line here please. and I think the second argument above is not aligned, but that could be github's display

File: src/hidpp20.h [hunk 3]
```diff
@@ -324,6 +332,12 @@ struct hidpp20_color {
 } __attribute__((packed));
 _Static_assert(sizeof(struct hidpp20_color) == 3, "Invalid size");
 
+enum hidpp20_led_type {
+	HIDPP20_LED_UNKNOWN = -1,
+	HIDPP20_LED_LOGO = 0,
+	HIDPP20_LED_SIDE
```
Comment (whot) [inline src/hidpp20.h]: add a trailing comma please, so when we add other things we don't need to change two lines

File: src/hidpp20.c [hunk 4]
```diff
@@ -1804,6 +1828,10 @@ int hidpp20_onboard_profiles_read(struct hidpp20_device *device,
 		profile->dpi[i] = hidpp_get_unaligned_le_u16(&data[2 * i + 3]);
 	}
 
+	for (i = 0; i < profiles_list->num_leds; i++) {
+		hidpp20_onboard_profile_read_led(&profile->leds[i], pdata.profile.leds[i]);
+	}
+
```
Comment (whot) [inline src/hidpp20.c]: just a nitpick: no {} for single-line if blocks

File: src/hidpp20.c [hunk 5]
```diff
@@ -1853,6 +1908,10 @@ int hidpp20_onboard_profiles_write(struct hidpp20_device *device,
 		pdata.profile.dpi[i] = hidpp_cpu_to_le_u16(profile->dpi[i]);
 	}
 
+	for (i = 0; i < profiles_list->num_leds; i++) {
+		hidpp20_onboard_write_profile_led(&pdata.profile.leds[i], &profile->leds[i]);
+	}
```
Comment (whot) [inline src/hidpp20.c]: same here, no {} for single-line blocks

**Comment by amfern** [comment]
Thanks @whot, fixed the styling issues.

**Comment by whot** [comment]
can you squash the squashme in please? :) Made it easier to check, I admit, so if you push this out squashed in I can merge it, thanks.

**Comment by amfern** [comment]
squash, thanks for the review

**Pull request merged**